DFC.TestAutomation.UI/Helper/BrowserHelper.cs

```diff
@@ -0,0 +1,67 @@
+﻿// <copyright file="BrowserHelper.cs" company="National Careers Service">
+// Copyright (c) National Careers Service. All rights reserved.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace DFC.TestAutomation.UI.Helper
+{
+    /// <summary>
+    /// Provides helper functions for all web browser related operations.
+    /// </summary>
+    public class BrowserHelper : IBrowserHelper
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BrowserHelper"/> class.
+        /// </summary>
+        /// <param name="browserName">The name of the browser.</param>
+        public BrowserHelper(string browserName)
+        {
+            this.BrowserName = browserName?.ToLower(CultureInfo.CurrentCulture).Trim();
+
+            if (!this.BrowserIndex.ContainsKey(this.BrowserName))
+            {
+                throw new InvalidOperationException($"Unable to initialise the BrowserHelper class as the browser '{this.BrowserName}' was not recognised.");
+            }
+        }
+
+        private string BrowserName { get; set; }
+
+        private Dictionary<string, BrowserType> BrowserIndex { get; } = new Dictionary<string, BrowserType>()
+        {
+            { "browserstack", BrowserType.BrowserStack },
+            { "cloud", BrowserType.BrowserStack },
+            { "zapProxyChrome", BrowserType.Chrome },
+            { "ie", BrowserType.InternetExplorer },
+            { "internetexplorer", BrowserType.InternetExplorer },
+            { "firefox", BrowserType.Firefox },
+            { "mozillafirefox", BrowserType.Firefox },
+            { "chrome", BrowserType.Chrome },
+            { "googlechrome", BrowserType.Chrome },
+            { "local", BrowserType.Chrome },
+            { "chromeheadless", BrowserType.Chrome },
+            { "headlessbrowser", BrowserType.Chrome },
+            { "headless", BrowserType.Chrome },
+        };
+
+        /// <summary>
+        /// Gets a BrowserType.
+        /// </summary>
+        /// <returns>A BrowserType based on the browser name.</returns>
+        public BrowserType GetBrowserType()
+        {
+            return this.BrowserIndex[this.BrowserName];
+        }
+
+        /// <summary>
+        /// Assesses whether the current test execution is cloud based. It uses the browser name to identify whether it is cloud based.
+        /// </summary>
+        /// <returns>True in the case where the current test execution is cloud based. False if the current execution is not cloud based.</returns>
+        public bool IsExecutingInTheCloud()
```

Comment:
@jbarnes16503 Think it would reduce maintenance if this function simply checks the `this.BrowserIndex[this.BrowserName]` = BrowserType.BrowserStack ....to remove the need for additional stringcomparisons to be added if extra 'cloud' terms added to the `Dictionary`.

Reply:
I have renamed this to IsExecutingInBrowserStack() as it is more explicit. I've updated the summary and I have applied the logic you have suggested. Thanks.